notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb

```diff
@@ -0,0 +1,853 @@
+    "* [Build a Vertex AI Pipeline](https://cloud.google.com/vertex-ai/docs/pipelines/build-pipeline)\n",
+    "\n",
+    "This notebook is built to run a Vertex AI User-Managed Notebook using the default Compute Engine Service Account. Check the Dataproc Serverless in Vertex AI Pipelines tutorial linked above to learn how to setup a different Service Account.\n",
+    "\n",
+    "## Permissions\n",
+    "Make sure that the service account used to run the notebook has the following roles:\n",
+    "\n",
+    "* roles/aiplatform.serviceAgent\n",
+    "* roles/aiplatform.customCodeServiceAgent\n",
+    "* roles/storage.objectCreator\n",
+    "* roles/storage.objectViewer\n",
+    "* roles/dataproc.editor\n",
+    "* roles/dataproc.worker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5e06a8de-c8b0-4ada-81c7-54606c0afdb9",
+   "metadata": {},
+   "source": [
+    "# Step 1: Install Libraries\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>NOTE: </b>Run Step 1 one time for each new notebook instance</div"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "258dae17-19f2-410a-8cc9-f5980ccc30f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip3 install pymssql SQLAlchemy\n",
+    "!pip3 install --upgrade google-cloud-pipeline-components kfp --user -q\n",
+    "!pip3 install google.cloud.aiplatform"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d5a7126d-fc9c-4f04-8031-cbd835f891c6",
+   "metadata": {},
+   "source": [
+    "### Once you've installed the additional packages, you may need to restart the notebook kernel so it can find the packages.\n",
+    "\n",
+    "Uncomment & Run this cell if you have installed anything from above commands"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "74245439-10f3-4631-b8cd-4c077318b737",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import os\n",
+    "# import IPython\n",
+    "# if not os.getenv(\"IS_TESTING\"):\n",
+    "#     app = IPython.Application.instance()\n",
+    "#     app.kernel.do_shutdown(True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50fe84b1-eb32-4cc2-a4e2-f0bd98da5b90",
+   "metadata": {},
+   "source": [
+    "# Step 2: Import Libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f225135-cfe5-4ddd-9659-ead437aa414c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import google.cloud.aiplatform as aiplatform\n",
+    "import sys\n",
+    "from kfp import dsl\n",
+    "from kfp.v2 import compiler\n",
+    "from datetime import datetime\n",
+    "import time\n",
+    "import copy\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "from google_cloud_pipeline_components.experimental.dataproc import DataprocPySparkBatchOp\n",
+    "import ipywidgets as widgets\n",
```

Comment:
error while importing ipywidgets

Reply:
Removed ipywidgets as its not being used anywhere in the notebook.